lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc

```diff
@@ -332,6 +335,16 @@ function dosomething_reportback_form_submit($form, &$form_state) {
     $user = user_load_by_name($form_state['values']['username']);
     $values['uid'] = $user->uid;
   }
+
+  // Load uid by selected username and give access to file.
+  if ($is_admin_form) {
+    $account = user_load_by_name($form_state['values']['username']);
+    $values['uid'] = $account->uid;
+
+    $reportback = reportback_load($values['rbid']);
+    $most_recently_added_photo_fid = end($reportback->fids);
+    $values['fid'] =  $most_recently_added_photo_fid;
```

Comment:
hmmm, plus as an admin I can still update Reportbacks on my own account when testing, so I feel the error would still happen.